### PR TITLE
validate: add validate app for travis automation

### PR DESCRIPTION
**Pull request by element-82**
The device-cloud-python has an automation script for CI.  Port that to
the clib version.  Update the .travis.yml file to call the script for
each build.  Add a README with some details.

Signed-off-by: Paul Barrette <paulbarrette@gmail.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,40 +1,47 @@
 sudo: false
 language: c
 compiler:
-  - clang
-  - gcc
+- clang
+- gcc
```
[the remark this answers is not in the document]
Comment (element-82) [reply .travis.yml]: my editor stuck them in.  I will strip the useless white space change.

File: .travis.yml [hunk 2]
```diff
@@ -1,40 +1,47 @@
...
+    - libjson-c-dev
+    - libcurl4-openssl-dev
 install:
-  - mkdir build
-  - cd build
-  - ../build.sh
-
+- mkdir build
+- sed -i "s|\(IOT_DEFAULT_DIR_CONFIG:\).*|\1 \"${PWD}/build/bin\"|" build.yml
+- sed -i "s|\(IOT_DEFAULT_DIR_RUNTIME:\).*|\1 \"${PWD}/build/bin\"|" build.yml
```
[the remark this answers is not in the document]
Comment (element-82) [reply .travis.yml]: I think once the bug is fixed, we can pass in the -D define on the cmake line.

File: share/apps/validate/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, either express or implied.
+
+add_sample_app( validate 
+	DISPLAY_NAME "Validate App"
+	DESCRIPTION "Simple validation application demonstrating actions"
+	README README.md
+	SRCS validate.c
+	TARGET iot-app-validate
+)
```
[the remark this answers is not in the document]
Comment (element-82) [reply share/apps/validate/CMakeLists.txt]: No, I just copied them in place in the .travis file.

File: share/apps/validate/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#
+# Copyright (C) 2017 Wind River Systems, Inc. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (element-82) [reply share/apps/validate/CMakeLists.txt]: Ack

File: share/apps/validate/validate.c [hunk 5]
```diff
@@ -0,0 +1,642 @@
...
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied."
+ */
+
+#include <iot.h>
+#include <sys/types.h>
+#include <sys/stat.h>
```
[the remark this answers is not in the document]
Comment (element-82) [reply share/apps/validate/validate.c]: I think I will scrub all windows data here.  This is meant for CI.  No requirement to run on windows.  I took that _WIN line from one of the other apps.

**Comment by lt-holman** [comment]
access to cloud has been disabled.. so this doesn't work anymore.

**Pull request closed without merging**